Utilities/bootstrap

```diff
@@ -681,7 +685,7 @@ def process_runtime_libraries(build, args, lib_path):
         cmd = [args.swiftc_path, "-emit-library", "-o", runtime_lib_path,
                "-Xlinker", "--whole-archive",
                "-Xlinker", input_lib_path,
-               "-Xlinker", "--no-whole-archive", "-lswiftGlibc",
+               "-Xlinker", "--no-whole-archive", "-lswiftGlibc", "-lFoundation",
```

Comment:
This is the only part of this pull that affects non-Android platforms. I had issues for a long time with getting the bootstrapped SPM to interpret its own manifest and dump it to JSON (as mentioned in #2396) and it turned out this was why.

The problem is that libPackageDescription.so depends on some functions from Foundation, but doesn't list it as a required dependency:
```
> readelf -sW 512-release/usr/lib/swift/pm/4_2/libPackageDescription.so | ag Foundation
   160: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $s10Foundation11JSONEncoderCACycfc
   161: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $s10Foundation11JSONEncoderCMa
   162: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $sSS10FoundationE4data8encodingSSSgAA4DataVh_SSAAE8EncodingVtcfC
   163: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $sSS10FoundationE8EncodingV4utf8ACvau
  1616: 000000000001d130    78 FUNC    LOCAL  HIDDEN    11 $s10Foundation4DataV15_RepresentationOWOe
  2341: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $s10Foundation11JSONEncoderCACycfc
  2342: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $s10Foundation11JSONEncoderCMa
  2445: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $sSS10FoundationE4data8encodingSSSgAA4DataVh_SSAAE8EncodingVtcfC
  2446: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND $sSS10FoundationE8EncodingV4utf8ACvau
> readelf -d 512-release/usr/lib/swift/pm/4_2/libPackageDescription.so| ag library
 0x0000000000000001 (NEEDED)             Shared library: [libswiftSwiftOnoneSupport.so]
 0x0000000000000001 (NEEDED)             Shared library: [libswiftCore.so]
 0x0000000000000001 (NEEDED)             Shared library: [libswiftGlibc.so]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000001d (RUNPATH)            Library runpath: [/home/buildnode/jenkins/workspace/oss-swift-5.1-package-linux-ubuntu-18_04/build/buildbot_linux/swift-linux-x86_64/lib/swift/linux:$ORIGIN/../../linux]
```
These results were from the official 5.1.2 release for linux x86_64, I get the same results with the latest snapshots from master.

The Android dynamic linker enforces that even if a executable or library links against libPackageDescription.so and lists libFoundation.so as a dependency there, it isn't transitively applied to libPackageDescription.so, unlike [the GNU dynamic linker historically, though that may be changing](https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1737608/comments/4). It appears that's why this hasn't hit on linux, as the Swift compiler seems to automatically add the needed Foundation dependency to a linux executable I tested linking against libPackageDescription.so, but that doesn't help on Android since it enforces that each library list its requirements itself.